src/test/java/io/jenkins/plugins/analysis/warnings/recorder/DashboardPortletITest.java

```diff
@@ -0,0 +1,204 @@
+import com.gargoylesoftware.htmlunit.WebRequest;
+
+import hudson.model.FreeStyleProject;
+import hudson.model.Project;
+import hudson.model.Result;
+import hudson.plugins.view.dashboard.Dashboard;
+
+import io.jenkins.plugins.analysis.core.testutil.IntegrationTestWithJenkinsPerSuite;
+import io.jenkins.plugins.analysis.warnings.recorder.pageobj.DashboardTable;
+
+import static org.assertj.core.api.Assertions.*;
+
+/**
+ * Integration tests for the dashboard portlet table.
+ *
+ * @author Michael Schmid, Raphael Furch
+ */
+public class DashboardPortletITest extends IntegrationTestWithJenkinsPerSuite {
+    /**
+     * Dashboard Portlet should show the recorded warning count according there configured tools.
+     */
+    @Test
+    public void showJobResultsAccordingTheirPluginConfiguration() {
+        FreeStyleProject project1 = createFreeStyleProjectWithWorkspaceFiles("checkstyle-filtering.xml",
+                "eclipse.txt");
+        enableEclipseWarnings(project1);
+        enableCheckStyleWarnings(project1);
+
+        FreeStyleProject project2 = createFreeStyleProjectWithWorkspaceFiles("checkstyle-healthReport.xml");
+        enableCheckStyleWarnings(project2);
+
+        Dashboard dashboard = configureDashboard("showJobResultsAccordingTheirPluginConfiguration", project1, project2);
+
+        buildWithResult(project1, Result.SUCCESS);
+        buildWithResult(project2, Result.SUCCESS);
+
+        DashboardTable pageObject = loadDashboardTable(dashboard);
+        assertThat(pageObject.getWarningCounts(project1.getName())).hasSize(2)
+                .containsEntry("CheckStyle", 7)
+                .containsEntry("Eclipse ECJ", 8);
+        assertThat(pageObject.getWarningCounts(project2.getName())).hasSize(1)
+                .containsEntry("CheckStyle", 6);
+    }
+
+    /**
+     * Dashboard Portlet should show the recorded warning count of the latest build.
+     */
+    @Test
+    public void showJobResultsOfTheLastBuild() {
+        FreeStyleProject project = createFreeStyleProjectWithWorkspaceFiles("checkstyle-filtering.xml", "eclipse.txt");
+        enableEclipseWarnings(project);
+        enableCheckStyleWarnings(project);
+
+        Dashboard dashboard = configureDashboard("showJobResultsOfTheLastBuild", project);
+        buildWithResult(project, Result.SUCCESS);
+
+        DashboardTable pageObject = loadDashboardTable(dashboard);
+        assertThat(pageObject.getWarningCounts(project.getName())).hasSize(2)
+                .containsEntry("CheckStyle", 7)
+                .containsEntry("Eclipse ECJ", 8);
+
+        copySingleFileToWorkspace(project, "checkstyle-healthReport.xml", "checkstyle-filtering-issues.txt");
+        buildWithResult(project, Result.SUCCESS);
+
+        pageObject = loadDashboardTable(dashboard);
+        assertThat(pageObject.getWarningCounts(project.getName())).hasSize(2)
+                .containsEntry("CheckStyle", 6)
+                .containsEntry("Eclipse ECJ", 8);
+    }
```

Comment:
It would also make sense to check that the links navigate to the correct results. The page objects are already there.